internal/resources/cluster/tkgaws/resource_tkg_aws.go

```diff
@@ -0,0 +1,967 @@
+				Description: "List of availability zones for the control plane nodes",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			instanceTypeKey: {
+				Type:        schema.TypeString,
+				Description: "Control plane instance type",
+				Required:    true,
+			},
+			highAvailabilityKey: {
+				Type:        schema.TypeBool,
+				Description: "Flag which controls if the cluster needs to be highly available. HA cluster creates three controlplane machines, and non HA creates just one",
+				Default:     false,
+				Optional:    true,
+			},
+		},
+	},
+}
+
+func expandTKGAWSTopologyControlPlane(data []interface{}) (controlPlane *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsControlPlane) {
+	if len(data) == 0 || data[0] == nil {
+		return controlPlane
+	}
+
+	lookUpControlPlane, _ := data[0].(map[string]interface{})
+	controlPlane = &tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsControlPlane{}
+
+	if avzones, ok := lookUpControlPlane[availabilityZonesKey]; ok {
+		avz, _ := avzones.([]interface{})
+
+		s := make([]string, 0)
+
+		for _, raw := range avz {
+			s = append(s, raw.(string))
+		}
+
+		controlPlane.AvailabilityZones = s
+	}
+
+	if v, ok := lookUpControlPlane[instanceTypeKey]; ok {
+		controlPlane.InstanceType, _ = v.(string)
+	}
+
+	if v, ok := lookUpControlPlane[highAvailabilityKey]; ok {
+		controlPlane.HighAvailability, _ = v.(bool)
+	}
+
+	return controlPlane
+}
+
+func flattenTKGAWSTopologyControlPlane(controlPlane *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsControlPlane) (data []interface{}) {
+	flattenControlPlane := make(map[string]interface{})
+
+	if controlPlane == nil {
+		return nil
+	}
+
+	flattenControlPlane[availabilityZonesKey] = controlPlane.AvailabilityZones
+	flattenControlPlane[instanceTypeKey] = controlPlane.InstanceType
+	flattenControlPlane[highAvailabilityKey] = controlPlane.HighAvailability
+
+	return []interface{}{flattenControlPlane}
+}
+
+var tkgAWSNodePoolInfo = &schema.Schema{
+	Type:     schema.TypeList,
+	Required: true,
+	MaxItems: 1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			nodePoolNameKey: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			nodePoolDescriptionKey: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	},
+}
+
+var tkgAWSNodePoolSpec = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Spec for the cluster nodepool",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			workerNodeCountKey: {
+				Type:        schema.TypeString,
+				Description: "Count is the number of nodes",
+				Optional:    true,
+			},
+			tkgAWSKey: {
+				Type:        schema.TypeList,
+				Description: "Nodepool config for tkg aws",
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						nodepoolAvailabilityZoneKey: {
```

Comment:
Nodepool schema is again specific to the version of the AWS workload cluster.
![image](https://user-images.githubusercontent.com/47659349/163361000-3dbcb4a0-c20b-41a8-9875-1dd45ec7d307.png)


Since we have `tkgAWSKey` as optional it should be fine